pydrive2/test/test_util.py

```diff
@@ -1,9 +1,60 @@
+import unittest
 import random
 import re
+import os
+
+from funcy import retry
+from funcy.py3 import cat
+from pydrive2.files import ApiRequestError
 
 newline_pattern = re.compile(r'[\r\n]')
 
 
+class PyDriveTestBase(unittest.TestCase):
+    GDRIVE_USER_CREDENTIALS_DATA = "GDRIVE_USER_CREDENTIALS_DATA"
+    DEFAULT_USER_CREDENTIALS_FILE = "credentials/1.dat"
+    if os.getenv(GDRIVE_USER_CREDENTIALS_DATA):
+        if not os.path.exists(os.path.dirname(DEFAULT_USER_CREDENTIALS_FILE)):
+            os.makedirs(os.path.dirname(DEFAULT_USER_CREDENTIALS_FILE), exist_ok=True)
+        with open(
+                DEFAULT_USER_CREDENTIALS_FILE, "w"
+        ) as credentials_file:
+            credentials_file.write(
+                os.getenv(GDRIVE_USER_CREDENTIALS_DATA)
+            )
+
+class PyDriveRetriableError(Exception):
+    pass
+
+
+# 15 tries, start at 0.5s, multiply by golden ratio, cap at 20s
+@retry(15, PyDriveRetriableError, timeout=lambda a: min(0.5 * 1.618 ** a, 20))
```

Comment:
@Suor can we use the simplified version from the latest DVC core PR here?

Reply:
Yes, we can. However, this is not a decorator, this is direct invoke with retries, so things need to be adjusted, or we should use a decorator here too, it is more universal anyway.